src/java/fr/paris/lutece/plugins/forms/web/FormXPage.java

```diff
@@ -566,6 +567,20 @@ public synchronized XPage doFormResponseSummary( HttpServletRequest request ) th
                 return getStepView(  request );
             }
             FormsResponseUtils.fillResponseManagerWithResponses( request, true, _formResponseManager, _stepDisplayTree.getQuestions( ), false );
+
+            List<Step> stepList = StepHome.getStepsListByForm( form.getId( ) );
```

Comment:
To avoid this database call, maybe we can add a method to FormResponseManager, wich already stores the form, to check for validated steps ?